Source/OpenPF2Core/Private/PF2RootCollisionDelegateComponent.cpp

```diff
@@ -46,27 +49,28 @@ bool UPF2RootCollisionDelegateComponent::MoveComponentImpl(const FVector&      D
 	this->ConditionalUpdateComponentToWorld();
 
 	// --- Start Difference from UPrimitiveComponent
-	this->CollisionComponent->ConditionalUpdateComponentToWorld();
+	Collision->ConditionalUpdateComponentToWorld();
 	// --- End Difference from UPrimitiveComponent
 
 	// Set up
 	const FVector TraceStart = this->GetComponentLocation();
 	const FVector TraceEnd   = TraceStart + Delta;
 
 	// --- Start Difference from UPrimitiveComponent
-	const FVector CollisionTraceStart = this->CollisionComponent->GetComponentLocation();
-	const FVector CollisionTraceEnd   = TraceStart + Delta;
-	// --- End Difference from UPrimitiveComponent
+	const FVector CollisionTraceStart  = Collision->GetComponentLocation();
+	const FVector CollisionTraceEnd    = CollisionTraceStart + Delta;
+
+	const FQuat InitialRotationQuat = Collision->GetComponentTransform().GetRotation();
 
 	// Recalc here to account for precision loss of float addition
-	float       DeltaSizeSq         = (TraceEnd - TraceStart).SizeSquared();
-	const FQuat InitialRotationQuat = this->CollisionComponent->GetComponentTransform().GetRotation();
+	float CollisionDeltaSizeSq = (CollisionTraceEnd - CollisionTraceStart).SizeSquared();
```

Comment:
I wasn't sure if it was strictly necessary to calculate this based on the collision trace, but it felt safer.